### PR TITLE
Add simple auth to all admin-only controllers

### PR DESCRIPTION
**Pull request by exbinary**
### Why
Gives all admin-only controllers rudimentary authorization by inheriting from `AdminController`.

See #835 for background.

### Pre-Merge Checklist
- [x] All new features have been described in the pull request
- [x] Security & accessibility have been considered
- [ ] High quality tests have been added, or an explanation has been given why the features cannot be tested
- [ ] All outstanding questions and concerns have been resolved
- [ ] Any next steps that seem like good ideas have been created as issues for future discussion & implementation

### Testing
This is part of the work to re-open user registration.
Once all that work is ready, we'll need to do some extensive manual regression testing.

### Specific review requests
Most importantly, please check whether all these controllers are actually only for admin use.

### Next Steps
?

### Outstanding Questions, Concerns and Other Notes
?

### CODE REVIEW

File: app/models/listing.rb [hunk 1]
```diff
@@ -67,7 +67,7 @@ def status
     elsif matches_as_provider.any?
       status = matches_as_provider.map{|m| m.completed?}.any? ? 'completed' : 'matched'
     end
-    update_attributes(state: status)
+    update(state: status)
```
Comment (exbinary) [inline app/models/listing.rb:70]: Gets rid of an annoying deprecation warning :)

File: spec/requests/listings_spec.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@
     location_attributes: { zip: '12e45' },
   }}
 
-  before { sign_in create(:user) }
+  before { sign_in create(:user, :admin) }
```
Comment (exbinary) [inline spec/requests/listings_spec.rb:16]: I believe `ListingController` is only currently used by admins?

**Comment by exbinary** [comment]
Pair reviewed with @maebeale .

**Review by maebeale — APPROVED** [review]
Perfecto! Some of these are likely to need to come out of this category, but until we have the relevant functionality, all of these should be restricted to admin/sys_admin only. Tks!

**Pull request merged**